### ISSUE TITLE
COMPlus_GCHeapHardLimit doesn't work

### ISSUE THREAD

**Issue by gao-artur** [dotnet/runtime#56270]
I'm trying to change the default 75% limit by setting `COMPLUS_GCHeapHardLimit` as environment variable but it's not respected by runtime. I have created test Worker project with following code:

```
private const long Mega = 1024 * 1024;

protected override async Task ExecuteAsync(CancellationToken stoppingToken)
{
    while (!stoppingToken.IsCancellationRequested)
    {
        try
        {
            var size = 1 * Mega;
            var chunk = new byte[size];

            for (var i = 0; i < chunk.Length; i++)
            {
                chunk[i] = (byte)(i % 256);
            }

            _garbage.Add(chunk);

            PrintMemoryUsage();
        }
        catch (Exception e)
        {
            _logger.LogError(e, "FAIL");
            throw;
        }

        await Task.Delay(100, stoppingToken);
    }
}

private void PrintMemoryUsage()
{
    var proc = Process.GetCurrentProcess();
    var totalAvailableMemoryBytes = GC.GetGCMemoryInfo().TotalAvailableMemoryBytes;
    _logger.LogInformation($"Working set {proc.WorkingSet64 / Mega} / {proc.MaxWorkingSet.ToInt64() / Mega} MB (TotalAvailableMemoryBytes: {totalAvailableMemoryBytes / Mega} MB)");
}
```

Create docker based on runtime:5.0-alpine
```
FROM mcr.microsoft.com/dotnet/runtime:5.0-alpine
```

And start the docker with memory limit of 600 MB and `COMPLUS_GCHeapHardLimit` 200 MB
```
docker run -it --rm --name MemoryLeak -m 600m -e COMPLUS_GCHeapHardLimit=C800000 memory-leak
```

The output
```
...
Working set 453 / 600 MB (TotalAvailableMemoryBytes: 450 MB)
Working set 454 / 600 MB (TotalAvailableMemoryBytes: 450 MB)
Working set 455 / 600 MB (TotalAvailableMemoryBytes: 450 MB)
Working set 456 / 600 MB (TotalAvailableMemoryBytes: 450 MB)
Working set 457 / 600 MB (TotalAvailableMemoryBytes: 450 MB)
FAIL
System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
     at MemoryLeak.Worker.ExecuteAsync(CancellationToken stoppingToken) in C:\repos\MemoryLeak\Worker.cs:line 45
```

Setting `System.GC.HeapHardLimit: 209715200` in `runtimeconfig.json` works as expected:
```
...
Working set 214 / 600 MB (TotalAvailableMemoryBytes: 200 MB)
Working set 216 / 600 MB (TotalAvailableMemoryBytes: 200 MB)
Working set 217 / 600 MB (TotalAvailableMemoryBytes: 200 MB)
Working set 218 / 600 MB (TotalAvailableMemoryBytes: 200 MB)
Working set 219 / 600 MB (TotalAvailableMemoryBytes: 200 MB)
FAIL
  System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
     at MemoryLeak.Worker.ExecuteAsync(CancellationToken stoppingToken) in C:\repos\MemoryLeak\Worker.cs:line 45
```

**Comment by gao-artur** [comment]
Oh! Found [this twit](https://twitter.com/maoni0/status/1141534998119342080) that solves my issue!
![image](https://user-images.githubusercontent.com/4518483/126880389-640d036a-7434-48d5-a7fe-223db5f8b82e.png)
The [documentation](https://docs.microsoft.com/en-us/dotnet/core/run-time-config/garbage-collector#heap-limit) says it should be `COMPLUS_GCHeapHardLimit` (note the `PLUS` in capital)

**Comment by Maoni0** [comment]
@gewarren, could we please make sure the casing is correct for all env vars in the [documentation](https://docs.microsoft.com/en-us/dotnet/core/run-time-config/garbage-collector)? the note on the COMPlus env vars at the beginning is correct about casing but in the tables they are all capitalized.

**Comment by mangod9** [comment]
Fixed by @gewarren in dotnet/docs#25341. Thanks!
